recipes/launchd_service.rb

```diff
@@ -15,19 +15,38 @@ class ::Chef::Recipe
 mac_service_supported = version_checker.satisfied_by?(Gem::Version.new(node['chef_packages']['chef']['version']))
 
 if mac_service_supported
-  template '/Library/LaunchDaemons/com.chef.chef-client.plist' do
-    source 'com.chef.chef-client.plist.erb'
-    mode 0644
-    variables(
-      :launchd_mode => node['chef_client']['launchd_mode'],
-      :client_bin => client_bin
-    )
+  if node['chef_client']['task']['user'].eql? 'SYSTEM'
+    template '/Library/LaunchDaemons/com.chef.chef-client.plist' do
+      source 'com.chef.chef-client.plist.erb'
+      mode 0644
+      variables(
+          :launchd_mode => node['chef_client']['launchd_mode'],
+          :client_bin => client_bin
+      )
+    end
+  else
+    home_dir = Dir.home(node['chef_client']['task']['user'])
+    
+    directory "#{home_dir}/Library/LaunchDaemons" do
+      owner node['chef_client']['task']['user']
```

Comment:
Have you seen the pivotal-sprout repos? They require use of sudo and will not run as root.

> On Mar 3, 2015, at 2:57 PM, Joshua Timberman <notifications@github.com> wrote:
> 
> In recipes/launchd_service.rb <https://github.com/opscode-cookbooks/chef-client/pull/279#discussion_r25725751>:
> 
> > -      :client_bin => client_bin
> > -    )
> > +  if node['chef_client']['task']['user'].eql? 'SYSTEM'
> > +    template '/Library/LaunchDaemons/com.chef.chef-client.plist' do
> > +      source 'com.chef.chef-client.plist.erb'
> > +      mode 0644
> > +      variables(
> > +          :launchd_mode => node['chef_client']['launchd_mode'],
> > +          :client_bin => client_bin
> > +      )
> > +    end
> > +  else
> > +    home_dir = Dir.home(node['chef_client']['task']['user'])
> > +    
> > +    directory "#{home_dir}/Library/LaunchDaemons" do
> > +      owner node['chef_client']['task']['user']
> If the intention here is to allow the chef-client service to run as a non-root user, is it worth supporting that use case? Some operations that one would want to do with Chef will require privileged access, or at least the use of sudo to provide privileged access. That won't work when run daemonized, since it will sit and wait for a password prompt...
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/opscode-cookbooks/chef-client/pull/279/files#r25725751>.
>